### ISSUE TITLE
Plot of time series with dashed and non-dashed parts.

### ISSUE THREAD

**Issue by fkroll** [vega/vega-lite#6911]
Hi together,

I wanted to replicated the example of "Multi-series Line Chart with Varying Dashes" where a line-plot is split into dashed and solid parts depending on a condition variable.

The following stage behaviour occurred when I tried to have multiple parts of dashed line-sections:

![visualization](https://user-images.githubusercontent.com/9720036/94569339-e87b0800-026d-11eb-9b08-61ff93597412.png)

The red rectangle marks the line that from my point of view should not be there.

Thanks for your help!

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "description": "Line chart with a dashed part created by drawing multiple connecting lines. Note that the data source contains the data point at (E, 81) twice.",
  "data": {
    "values": [
      {"a": "A", "b": 28, "predicted": false},
      {"a": "B", "b": 55, "predicted": false},
      {"a": "D", "b": 91, "predicted": false},
      {"a": "E", "b": 81, "predicted": false},
      {"a": "E", "b": 81, "predicted": true},
      {"a": "G", "b": 19, "predicted": true},
      {"a": "H", "b": 87, "predicted": true}, 
      {"a": "H", "b": 87, "predicted": false},
      {"a": "W", "b": 91, "predicted": false},
      {"a": "Z", "b": 91, "predicted": false}
    ]
  },
  "mark": "line",
  "encoding": {
    "x": {"field": "a", "type": "ordinal"},
    "y": {"field": "b", "type": "quantitative"},
    "strokeDash": {"field": "predicted", "type": "nominal"}
  }
}
```

[Vega-Lite Editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JACyUAVhDYA7EABoQAEzjQATjRyZ289ABkasuAAJYSdZhMB3OjBNITypBHjKTOC1ajq4STHDuTACeTupItrJkJggMgjo4gqZQcsZQOlEmwsYQlCYAcmwBJvT+JfBO-o4yDOpQyXKYSEYQ5abOTR5sRlZlABQAogomABwAjACUJbb1lIoqVWigxEiCDGpoANqgKOgAgvNMaABMI0o4vso06YFoAGarEHAAvgo7aCAAQodoAKy-50u1wCynujxebxAuxAABEfqgAJxjQGBYG3VAPQRPV7vdADeHjFFXG6gjHgnFQj74pRHVCEkAXVEktCYdTrCnQgDi8LGCKJaNJrPZkOhAAkCQB2fnM1BCiG4kDimloEZShlAmWY7Eij4AdXhSOlILBWPllPQAC0Dcj1UzjWTTc8ALqvEDIdQAaw+2Tg8zgshSVyiSxAAA8Q3caHBBKTKUpMMEcL70Gx1EHViBXcEI1GYx8jvHE8mQABHBhIWQ6Jo6UiZpQQVlsD1wGEuGA56OxxnE42FpMfWRsBBGDPPMdAA)

**Comment by domoritz** [comment]
This is the expected behavior. Encoding with strokeDash facets the data and creates two lines. If you want to change the stroke dash for part of the line, you can split the line in three parts or add null values for the solid line where you want it to skip a part (https://vega.github.io/vega-lite/docs/line.html#line-detail).

**Comment by fkroll** [comment]
Thank you for your answer! 

I tried both suggested solutions and added the results in case someone has the same "issue":

1. The one that splits the line in [ordered segments](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JACyUAVhDYA7EABoQAEzjQATjRyZ289ABkasuAAJYSdZhMB3OjBNITypBHjKTOC1ajq4STHDuTACeTupItrJkJggMgjo4gqZQcsZQOlEmwsYQlCYAcmwBJvT+JfBO-o4yDOpQpr4Ajgw0vhAlbCYwSKRZSExwSe4Q5Ahwspi5iipVaKDESIIMamgA2qAo6ACC00xoAEwAHEojZGMTaCAoSji+yjTpgWgAZosjAL4KG5cAQrtoAFYASdRuNMJdriBboEHgFlC83nBPt90AARf6oACcAEYQWcwRDptD7o94ahXoIPl8rpcAKIYw64kCnc7g9CQ4mwp7kxHImnoABiGOxmLxrMuexudy5ZMw6mWfM2IAA4hiAMyi5mgi7oSVQ6WktByhXUpUASQxYgADGKCbqiQa4Ub5UjTZcAFIYgBsAHZbTqQFAHTDDTzKa6USB9BifTatfiA0GpSGnWGqZGALLq-b+tkqYMk1PGiP8kAANUtfvj4vQ8OThe5xcVlwAmhiAXGWXb8-WZc6FQBdT4gZDqADWl2ycGm4xS9yicxAAA9F88aIMyTSlJhgjhp+g2Op54sQMPgqv14JN3qd3vLs0kBM6P4aKRTyc5Wwx3BUS4YBeN0uTlDW3Xd9xAWQ2AQIwT2HVRMCQGhBAAq9Li7C53kwoA ).
2.  And the one that [inserts duplicate and null values ](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JACyUAVhDYA7EABoQAEzjQATjRyZ289ABkasuAAJYSdZhMB3OjBNITypBHjKTOC1ajq4STHDuTACeTupItrJkJggMgjo4gqZQcsZQOlEmwsYQlCYAcmwBJvT+JfBO-o4yDOpQyXKYSEYQ5abOTR5sRpitZQAUAKIKJgAcAIwAlCP9ABIjYgAM0yb9AFIjAOzLDrLu-fkLm5MltvWUiipVaKDESIIMamgA2qAo6ACCl0xoAEyjShwvmUNHSgTQADN7hA4ABfBRvNAgABC3zQAFZ0YDgaCAspIdC4QiQO8QAARNGoACc42xgVx4NQUMEMPhiPQg0pEzpILB+KZhLZJKRnKUP1Qsjigh5DP5zNZxNJopA4u5ICB9L5aEw6keQtJAHFKQBmKkyrWoHV6xVI2aUpbmvHa3VE9kgO1itCSwTS9U4i1W13C9AelVoB1+zVOgUsoOkgCSlIAbJtHYz5XGkWtKds03LBTb0NnPRKpXmCbH9Vmc4ty5aXVX0ABZE2-OuBxsgfL21OR3nRjuFkAAeUp6NrfdlzutbtHJe9vo1-cZg9nY4nS6nMYVboAismN-7oxnOwAvHNYycWk8AXXhIEESGCcHUL1AyHUAGskdk4Jc4LIKQglENwgAAHqBEI0HAgj8sKSiYMEOB-ugbDqMB9wgPewSQdBsFIj8CFIShIAAI4MEgsg6E0OikFhSgQDqbCfnAZIuDAuEwXBm5akRyFIrIbAIEYmGwmJd5AA) into the data.

Now I understand why this is an expected behaviour in Vega-Lite and that it makes sense in the bigger picture.
But I think with this in mind, that the example in the documentation fools the novel Vega-Lite user, like me.
My first impression was that I am able to label a single line.
-> It would be nice if this can be clarified in the documentation by adding a more complex example similar to my examples that mentions that with the correct data structure this specific result can be achieved.

Last but not least.
Is there a Vega-Lite way to transform a time-series data set (like the one in the first example) without the need to insert the null-values and duplicate-points? Is there any Vega-functionality that I can look into and utilise for my purpose?
I tried several Transformations (Window Transform, Calculate Transform etc.) but finding that none are able to deliver the desired result. 
 
The idea is to receive segment labels that can be used in the "detail" parameter:
Get from this data with Vega-functionality ... 

```
"data": {
    "values": [
      {"a": "A", "b": 28, "predicted": false},
      {"a": "B", "b": 55, "predicted": false},
      {"a": "D", "b": 91, "predicted": false},
      {"a": "E", "b": 81, "predicted": false},
      {"a": "F", "b": 19, "predicted": true},
      {"a": "G", "b": 39, "predicted": true},
      {"a": "I", "b": 40, "predicted": true},
      {"a": "J", "b": 67, "predicted": false},
      {"a": "L", "b": 70, "predicted": false},
      {"a": "M", "b": 32, "predicted": true},
      {"a": "V", "b": 47, "predicted": true},
      {"a": "Y", "b": 50, "predicted": true}
    ]
  }
```

…to this data...

```
"data": {
    "values": [
      {"a": "A", "b": 28, "segment": "a", "predicted": false},
      {"a": "B", "b": 55, "segment": "a", "predicted": false},
      {"a": "D", "b": 91, "segment": "a", "predicted": false},
      {"a": "E", "b": 81, "segment": "a", "predicted": false},
      {"a": "F", "b": 19, "segment": "b", "predicted": true},
      {"a": "G", "b": 39, "segment": "b", "predicted": true},
      {"a": "I", "b": 40, "segment": "b", "predicted": true},
      {"a": "J", "b": 67, "segment": "c", "predicted": false},
      {"a": "L", "b": 70, "segment": "c", "predicted": false},
      {"a": "M", "b": 32, "segment": "d", "predicted": true},
      {"a": "V", "b": 47, "segment": "d", "predicted": true},
      {"a": "Y", "b": 50, "segment": "d", "predicted": true}
    ]
  }
```

I would like to achieve the results in Vega itself because I like to avoid to manipulate and "hack" my data-structure just to match a single plot.
For my use case I work with large DataFrames (60,000+ rows) and I really like to avoid to inserting null and duplicate values.

I hope my examples could help to clarify the documentation and thanks for your help!

**Comment by domoritz** [comment]
Thank you for the follow up and providing the examples! 

> Is there a Vega-Lite way to transform a time-series data set (like the one in the first example) without the need to insert the null-values and duplicate-points? Is there any Vega-functionality that I can look into and utilise for my purpose?
I tried several Transformations (Window Transform, Calculate Transform etc.) but finding that none are able to deliver the desired result.


Is https://vega.github.io/vega-lite/docs/impute.html what you are looking for?

**Comment by fkroll** [comment]
@domoritz : I have tried to use impute but it does not help, since it I was not able to label my true-false-segements in the order of column "a" (or in my case the time). And it just replaces missing values if they are missing. Maybe there is another trick I do not know. 
I think the problem is, that these true-false-segment-labels (see column "segment" in my data-example) are dependent on the order of another row. Thus, one needs this information and must iterate with a window of two over the "predicted" column's rows in that order. I do not see any way to do the same with Vega-Lite transformations. But it was worth the try. 

So for my real-life use-case I went with generating the true-false-segment-labels with a script on a sub-set of my data.
And then I can pass these labels to the "detail" property.
